lib/opencensus_phoenix/instrumenter.ex

```diff
@@ -58,12 +62,57 @@ defmodule OpencensusPhoenix.Instrumenter do
     :ocp.with_span_ctx(parent_span_ctx)
   end
 
-  def span_name(conn) do
+  defp span_name(conn, :controller_action, _) do
+    controller_action(conn)
+  end
+
+  defp span_name(conn, :route_name, %{route: route}) do
+    conn.method <> " " <> route
+  end
+
+  defp span_name(conn, :route_name, nil) do
+    log_route_name_unsupported()
+
+    controller_action(conn)
+  end
+
+  defp span_name(conn, {module, function}, _) do
+    apply(module, function, [conn])
+  end
+
+  defp span_name(conn, {module, function, args}, _) do
+    apply(module, function, [conn | args])
+  end
+
+  defp span_name(conn, _, _) do
+    controller_action(conn)
+  end
+
+  defp controller_action(conn) do
     controller = Phoenix.Controller.controller_module(conn)
     action = Phoenix.Controller.action_name(conn)
     "#{controller}.#{action}"
   end
 
+  defp route_info(%{method: method, request_path: request_path, host: host} = conn) do
+    route_info_present? = :erlang.function_exported(Phoenix.Router, :route_info, 4)
+
+    if route_info_present? do
+      router = conn.private[:phoenix_router]
+
+      apply(Phoenix.Router, :route_info, [router, method, request_path, host])
+    else
+      nil
+    end
+  end
+
+  defp log_route_name_unsupported do
+    Logger.debug("""
```

Comment:
Could we log this only if `Application.get_env(:opencensus_phoenix, :warn_on_route_name_fallback, true)`? Strikes me it could get spammy.

Reply:
Yeah, I wasn't even really sure if we should warn here at all. If its in the documentation that pre 1.4 will always have to fallback, I'm not even sure its worth doing this.

Reply:
+1 to documenting the expected span name by Phoenix version and removing `log_route_name_unsupported/0` entirely.

Reply:
Documented, and logging removed. Much better this way, agreed.